packages/kit/src/runtime/server/page/load_data.js

```diff
@@ -149,7 +150,11 @@ export async function load_data({
 							fetched.push({
 								url: same_origin ? url.href.slice(event.url.origin.length) : url.href,
 								method: event.request.method,
-								request_body: /** @type {string | ArrayBufferView | undefined} */ (init?.body),
+								request_body: /** @type {string | ArrayBufferView | undefined} */ (
+									input instanceof Request && cloned_body
+										? await streamToString(cloned_body)
+										: init?.body
```

Comment:
one thought — if `request_body` could be a `ReadableStream`...

```suggestion
									init?.body ?? input.clone?.().body
```

...then later, in `render.js`, we could do something like this:

```diff
	if (page_config.ssr && page_config.csr) {
+		for (const request of fetched) {
+			if (fetched.request_body instanceof ReadableStream) {
+				fetched.request_body = await to_string(fetched.request_body);
+			}
+		}
		body += `\n\t${fetched
			.map((item) =>
				serialize_data(item, resolve_opts.filterSerializedResponseHeaders, !!state.prerendering)
			)
			.join('\n\t')}`;
	}
```

Though it may actually be cheaper to keep the entire request clone and then do

```js
fetched.request_body = await fetched.clone.arrayBuffer();
```

than to implement our own stream to string function

Reply:
That doesn't work, because cloning the request at that point is too late, the body's `ReadableStream` is already consumed by then and you can't get the value.

Reply:
oh, of course 😞